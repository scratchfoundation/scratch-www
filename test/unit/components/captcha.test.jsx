const React = require('react');
const enzyme = require('enzyme');


const Captcha = require('../../../src/components/captcha/captcha.jsx');

describe('Captcha test', () => {
    global.grecaptcha = {
        execute: jest.fn(),
        render: jest.fn()
    };

    test('Captcha load calls props captchaOnLoad', () => {
        const props = {
            onCaptchaLoad: jest.fn()
        };
        const wrapper = enzyme.shallow(<Captcha
            {...props}
        />);
        wrapper.instance().onCaptchaLoad();
        expect(global.grecaptcha.render).toHaveBeenCalled();
        expect(props.onCaptchaLoad).toHaveBeenCalled();
    });

    test('Captcha execute calls grecatpcha execute', () => {
        const props = {
            onCaptchaLoad: jest.fn()
        };
        const wrapper = enzyme.shallow(<Captcha
            {...props}
        />);
        wrapper.instance().executeCaptcha();
        expect(global.grecaptcha.execute).toHaveBeenCalled();
    });

    test('Captcha renders the div google wants', () => {
        const props = {
            onCaptchaLoad: jest.fn()
        };
        const wrapper = enzyme.mount(<Captcha
            {...props}
        />);
        expect(wrapper.find('div.g-recaptcha')).toHaveLength(1);
    });
});
