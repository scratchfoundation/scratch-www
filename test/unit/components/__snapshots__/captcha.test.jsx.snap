// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Captcha test Captcha renders the div google wants 1`] = `
<div
  class="g-recaptcha"
  data-badge="bottomright"
  data-size="invisible"
/>
`;
